packages/docs/src/components/docsearch/doc-search.tsx

```diff
@@ -110,27 +122,35 @@ export const DocSearch = component$((props: DocSearchProps) => {
           if (searchButtonRef && searchButtonRef.value === document.activeElement) {
             if (/[a-zA-Z0-9]/.test(String.fromCharCode(event.keyCode))) {
               state.isOpen = true;
+              isOpen.value = true;
               state.initialQuery = event.key;
             }
           }
         }),
       ]}
     >
-      <DocSearchButton
-        ref={searchButtonRef}
-        onClick$={() => {
-          state.isOpen = true;
-        }}
-      />
-      {state.isOpen && (
-        <DocSearchModal
-          aiResultOpen={aiResultOpen.value}
-          indexName={props.indexName}
-          apiKey={props.apiKey}
-          appId={props.appId}
-          state={state}
-        />
-      )}
+      <Modal.Root bind:show={isOpen}>
+        <Modal.Trigger
+          class="DocSearch DocSearch-Button"
+          onClick$={() => {
+            state.isOpen = true;
+            isOpen.value = true;
+          }}
+        >
+          Search
+        </Modal.Trigger>
+        <Modal.Panel class="w-full h-full">
+          {state.isOpen && (
```

Comment:
Actually if you feel up to it, it would be cool to change the docsearchmodal so it uses sync$ instead of all those noSerialize things and then the search box will be instantly available

Reply:
the noSerialize  scope over some  variables (state)  and from what I read  that a no good? i am missing something  👀